### PR TITLE
[`Llama`] remove prompt and fix prefix finetuning

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?

We are waiting for #25323 to be merged, but following the llama update we are going to remove the default system prompt.

### CODE REVIEW

File: src/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -683,7 +683,7 @@ def forward(
                 def create_custom_forward(module):
                     def custom_forward(*inputs):
                         # None for past_key_value
-                        return module(*inputs, output_attentions, None)
```
Comment (ArthurZucker) [inline src/transformers/models/llama/modeling_llama.py:686]: the order was wrong

File: src/transformers/models/llama/modeling_llama.py [hunk 2]
```diff
@@ -692,7 +692,6 @@ def custom_forward(*inputs):
                     hidden_states,
                     attention_mask,
                     position_ids,
-                    None,
                 )
```
Comment (ArthurZucker) [inline src/transformers/models/llama/modeling_llama.py:695]: should be output_attention.
```suggestion
                    output_attentions,
```

**Review by sgugger — APPROVED** [review]
LGTM, thanks!

**Comment by ArthurZucker** [comment]
cc @sourabh112 as I am not an expert in prefix tuning, there was something reported in the llama repo: https://github.com/facebookresearch/llama-recipes/issues/47#issuecomment-1674492457

**Pull request merged**